build_projects/dotnet-host-build/PrepareTargets.cs

```diff
@@ -42,6 +42,17 @@ public class PrepareTargets
         public static BuildTargetResult Init(BuildTargetContext c)
         {
             var configEnv = Environment.GetEnvironmentVariable("CONFIGURATION");
+            string platformEnv = Environment.GetEnvironmentVariable("TARGETPLATFORM") ?? RuntimeEnvironment.OperatingSystemPlatform.ToString();
+            string targetRID = Environment.GetEnvironmentVariable("TARGETRID");
+            if (targetRID == null)
+            {
+                targetRID = RuntimeEnvironment.GetRuntimeIdentifier();
+                if (targetRID.StartsWith("win") && (targetRID.EndsWith("x86") || targetRID.EndsWith("x64")))
```

Comment:
If the TARGETRID env variable isn't set in build.ps1 (i.e. the TargetArch isn't set), then `RuntimeEnvironment.GetRuntimeIdentifier` returns your version-specific RID e.g. on a win10 x64 machine it returns `win10-x64`. We want it to be `win7-x64`.